sklearn/preprocessing/tests/test_target_encoder.py

```diff
@@ -712,3 +714,216 @@ def test_pandas_copy_on_write():
     with pd.option_context("mode.copy_on_write", True):
         df = pd.DataFrame({"x": ["a", "b", "b"], "y": [4.0, 5.0, 6.0]})
         TargetEncoder(target_type="continuous").fit(df[["x"]], df["y"])
+
+
+# ---------------------------------------------------------------------------
+# Small batch Target encoder tests
+# ---------------------------------------------------------------------------
+
+
+def _fit_pair_numpy():
+    """
+    Fit two encoders on the same tiny dataset:
+    - te_fast uses the small-batch fast-path (threshold >= 1024),
+    - te_vec forces the baseline vectorized path (threshold < 0).
+    """
+    X_fit = np.array(
+        [
+            ["u", "x"],
+            ["v", "y"],
+            ["u", "x"],
+            ["w", "y"],
+        ],
+        dtype=object,
+    )
+    y = np.array([0.0, 1.0, 1.0, 0.0])
+    te_fast = TargetEncoder(smooth=5.0).fit(X_fit, y)
+    # White-box: enable small-batch fast path deterministically.
+    te_fast._small_batch_threshold = 1024
+    te_vec = TargetEncoder(smooth=5.0).fit(X_fit, y)
+    # White-box: force always-vectorized path for reference.
+    te_vec._small_batch_threshold = -1
+    return te_fast, te_vec
+
+
+def _force_slow_transform(enc: TargetEncoder, X):
+    """Force the reference (vectorized) transform path by disabling small-batch"""
+    old_thresh = enc._small_batch_threshold
+    enc._small_batch_threshold = -1
+    try:
+        return enc.transform(X)
+    finally:
+        enc._small_batch_threshold = old_thresh
+
+
+# ------------------------------
+# Tests: NA/NAT normalization
+# ------------------------------
+
+
+def test_norm_key_real_values_cover_nan_nat_and_strings():
+    """
+    `_norm_key` should:
+    - map None, float NaN, and NumPy NaT to *distinct* sentinels,
+    - return ordinary strings unchanged.
+    """
```

Comment:
```suggestion
def test_norm_key_real_values_cover_nan_nat_and_strings():
    """
    `_norm_key` should:
    - map None, float NaN, and NumPy NaT to *distinct* sentinels,
    - return ordinary strings unchanged.
    """
```
Please also update this docstring. It should describe briefly in 1 or 2 sentences what the test is doing.